crates/linker/src/pe.rs

```diff
@@ -387,9 +387,30 @@ pub(crate) fn surgery_pe(executable_path: &Path, metadata_path: &Path, roc_app_b
                         destination - section_virtual_address as i64 - *offset_in_section as i64
                             + relocation.addend();
 
+                    executable[offset + *offset_in_section as usize..][..4]
+                        .copy_from_slice(&(delta as i32).to_le_bytes());
+                } else if name == "___chkstk_ms" {
+                    // this is a stack probe that is inserted when a function uses more than 2
+                    // pages of stack space. The source of this function is not linked in, so we
+                    // have to get a bit creative: we just jump to a `ret` instruction, so this
+                    // function call becomes a no-op.
```

Comment:
a bit hacky, but probably fine?

Reply:
[This article](https://metricpanda.com/rival-fortress-update-45-dealing-with-__chkstk-__chkstk_ms-when-cross-compiling-for-windows/) seems to be saying the app can crash.

> A problem arises when a function uses more than 2 pages for its stack variables. This means that it could possibly access memory past the guard page, thus triggering an access violation that won’t be handled by the OS as a simple request for more stack space, but as a generic exception that would terminate the application.

I suppose the way to test if it's a problem would be to create a function with a massive struct in it. Maybe a Zig or C function with a big array on the stack?

If it's a problem then I guess we need the source for the routine, which in theory could be gotten by compiling a pure Zig or C program to see what they generate. Then presumably we would need to store that in an array of bytes and insert it? Maybe using `include_bytes!`?

Reply:
Also, is this the only thing that makes our PE linker x86-specific, or is that assumption already made in lots of places?

Reply:
I think it is the only thing? of course including and adding assembly would also make it specific.